app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -9,31 +9,94 @@ module ClassMethods
     def validate_config_info(options)
       config_info = options[:config_info]
 
+      # ------------------------------------------------------------------------
+      # TransformationMapping validation
+      # ------------------------------------------------------------------------
       mapping = if config_info[:transformation_mapping_id]
                   TransformationMapping.find(config_info[:transformation_mapping_id])
                 else
                   config_info[:transformation_mapping]
                 end
-
+      # mapping.invalid? added as a part of validating migration plan
       raise _('Must provide an existing transformation mapping') if mapping.blank?
+      unless mapping.valid?
+        raise _('Transformation mapping was invalid.  ' + mapping.errors.messages)
+      end
 
+      # ------------------------------------------------------------------------
+      # check for openstack as a target for Flavors and SecurityGroup validation
+      # ------------------------------------------------------------------------
+      mapping_item        = mapping.transformation_mapping_items.find_by(:destination_type => "CloudTenant")
+      cloud_tenant        = mapping_item.destination if mapping_item.present?
```

Comment:
If `mapping_item` isn't used elsewhere you can just do `cloud_tenant = mapping.transformation_mapping_items.find_by(:destination_type => "CloudTenant")&.destination`